pkg/occlient/occlient.go

```diff
@@ -107,6 +118,63 @@ func runOcComamnd(command *OcCommand) ([]byte, error) {
 	return output, nil
 }
 
+func isLoggedIn() bool {
+	ocpath, err := getOcBinary()
+	if err != nil {
+		log.Debug(errors.Wrap(err, "unable to find oc binary"))
+		return false
+	}
+
+	cmd := exec.Command(ocpath, "whoami")
+	output, err := cmd.CombinedOutput()
+	if err != nil && strings.Contains(string(output), "system:anonymous") {
+		log.Debug(errors.Wrap(err, "error running command"))
+		log.Debugf("Output is: %v", output)
+		return false
+	}
+	return true
+}
+
+func isServerUp() bool {
+
+	ocpath, err := getOcBinary()
+	if err != nil {
+		log.Debug(errors.Wrap(err, "unable to find oc binary"))
+		return false
+	}
+
+	cmd := exec.Command(ocpath, "whoami", "--show-server")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Debug(errors.Wrap(err, "error running command"))
+		return false
+	}
+
+	server := strings.TrimSpace(string(output))
+	u, err := url.Parse(server)
+	if err != nil {
+		log.Debug(errors.Wrap(err, "unable to parse url"))
+		return false
+	}
+
+	var connectionError error
+	for i := 1; i < 3; i++ {
+		_, connectionError = net.DialTimeout("tcp", u.Host, time.Duration(ocRequestTimeout))
```

Comment:
You don't need to manually check this like this. 
`oc whoami` is communicating with the sever side so you can use just that.

```
▶ oc whoami
Unable to connect to the server: dial tcp 192.168.64.5:8443: i/o timeout
```

Reply:
`oc whoami` might also fail for different reasons, e.g. if a user is not logged in i.e. the user is `system:anonymous`, it's going to fail.

Reply:
yes, but the message that it outputs is pretty descriptive :-)

Reply:
I think we want to hide that message, which is why :o

Reply:
ok.